src/constructor-lock.ts

```diff
@@ -1,13 +1,5 @@
 /**
  * Used to enforce illegal constructors
  */
-let lock = true;
-
-export function setLock(value: boolean) {
-  lock = value;
-}
-
-export function getLock(): boolean {
-  return lock;
-}
 
+export const CTOR_KEY = {};
```

Comment:
Maybe `Symbol` would be better?  It would have the same functionality but `Symbol` was sorta made for this.

Reply:
Thanks.
If @b-fuze shows interest in this PR I'll definitely do this.

Reply:
Yeah, honestly, I was thinking of whether there were also any alternatives... But honestly, this makes the most sense, since if an illegal constructor is suddenly legal then this will fairly straightforward to patch. I'm wondering what @0kku thinks

Reply:
The only alternative I could see was building instances using the old style Object.create.  
It works well but I'm not very comfortable with it.  
A minimal example:  
```
class Element {
  constructor() {
    throw new DOMException("Invalid constructor");
  }

  initialize(name) {
    this.name = name;
    console.log("initialize", name);
  }

  static create(...args) {
    const element = Object.create(Element.prototype);
    Element.prototype.initialize.apply(element, args);
    return element;
  }
}
const element = Element.create("div");
element instanceof Element; // true
element.name; // div
```